node/service_setup.go

```diff
@@ -101,10 +75,17 @@ func (node *Node) StopServices() {
 	node.serviceManager.StopServicesByRole([]service.Type{})
 }
 
-func (node *Node) networkInfoDHTPath() string {
-	return fmt.Sprintf(".dht-%s-%s-c%s",
-		node.SelfPeer.IP,
-		node.SelfPeer.Port,
-		node.chainConfig.ChainID,
-	)
+// ForceJoiningTopics ..
+func (node *Node) ForceJoiningTopics() error {
+	groups := []nodeconfig.GroupID{
+		node.NodeConfig.GetShardGroupID(),
+		nodeconfig.NewClientGroupIDByShardID(shard.BeaconChainShardID),
+		node.NodeConfig.GetClientGroupID(),
+	}
+
+	// force the side effect of topic join
```

Comment:
elaborate a bit more? What side effect?

Reply:
there is a nuanced behavior of pubsub that under gossip sub, you must first publish before subscribe works (this is only when pubsub running as gossip sub)